File: 2021/1/b.py
with open('input.txt') as file:
    prev_4 = []
    increased = 0
    while line := file.readline().strip():
        cur = int(line)
        prev_4.append(cur)
        if len(prev_4) > 4:
            prev_4.pop(0)
        if len(prev_4) < 4:
            continue
        sum_1 = sum(prev_4[:-1])
        sum_2 = sum(prev_4[1:])
        if sum_2 > sum_1:
            increased += 1
    print(increased)

File: 2021/1/b.rs
use std::env;
use std::fs;
use std::collections::VecDeque;

fn main() {
    let args: Vec<String> = env::args().collect();
    let filename = &args[1];

    let contents = fs::read_to_string(filename)
        .expect("Could not read file");
    
    let mut increments: i32 = 0;
    let mut last_4 = VecDeque::with_capacity(4);
    for line in contents.lines() {
        let cur = line.parse::<i32>().unwrap();
        last_4.push_back(cur);
        if last_4.len() > 4 {
            last_4.pop_front();
        }
        if last_4.len() < 4 {
            continue;
        }
        let sum: i32 = last_4.iter().sum();
        let sum_1 = sum - last_4[3];
        let sum_2 = sum - last_4[0];
        if sum_2 > sum_1 {
            increments += 1;
        }
    }
    println!("{}", increments);
}
